packages/suite/src/utils/wallet/coinmarket/buyUtils.ts

```diff
@@ -120,19 +121,25 @@ export const getStatusMessage = (status: BuyTradeStatus) => {
 export const getCryptoOptions = (
     symbol: Account['symbol'],
     networkType: Account['networkType'],
+    supportedCoins: Set<string>,
+    coinInfo: AppState['wallet']['coinmarket']['exchange']['exchangeCoinInfo'],
 ) => {
-    const supportedTokens = ['usdt', 'usdc', 'dai', 'gusd'];
     const uppercaseSymbol = symbol.toUpperCase();
     const options: { value: string; label: string }[] = [
         { value: uppercaseSymbol, label: uppercaseSymbol },
     ];
 
     if (networkType === 'ethereum') {
-        supportedTokens.forEach(token => {
-            options.push({
-                label: token.toUpperCase(),
-                value: symbolToInvityApiSymbol(token).toUpperCase(),
-            });
+        // cycle through all coins, locate ERC20 tokens and if it is in supportedCoins, add it as option
+        coinInfo?.forEach(coin => {
+            if (coin.category === 'Ethereum ERC20 tokens') {
+                const ticker = coin.ticker.toLowerCase();
+                if (supportedCoins.has(ticker))
+                    options.push({
+                        label: invityApiSymbolToSymbol(ticker).toUpperCase(),
+                        value: coin.ticker,
+                    });
```

Comment:
I'd rather explicitly wrap the if block with braces `{}`

Reply:
Done